apps/emqx_durable_storage/src/emqx_ds_sup.erl

```diff
@@ -28,19 +28,22 @@
 %% Type declarations
 %%================================================================================
 
--define(SUP, ?MODULE).
+-define(TOP, ?MODULE).
 -define(TAB, ?MODULE).
 
+-define(WATCH_SUP, emqx_ds_new_streams_watch_sup).
+
 %%================================================================================
 %% API functions
 %%================================================================================
 
 -spec start_link() -> {ok, pid()}.
 start_link() ->
-    supervisor:start_link({local, ?SUP}, ?MODULE, top).
+    supervisor:start_link({local, ?TOP}, ?MODULE, top).
 
 register_db(DB, Backend) ->
     ets:insert(?TAB, {DB, Backend}),
+    _ = supervisor:start_child(?WATCH_SUP, [DB]),
```

Comment:
How is this supervisor stopped on db close/drop?

Reply:
It's never stopped by design. Stopping it would lose subscriptions and all events. I'll add a comment.